corehq/tests/nose.py

```diff
@@ -304,7 +340,12 @@ def teardown(self):
             return
 
         self.blob_db.close()
-        self.optimizer.__exit__(None, None, None)
+        if self.optimize:
+            self.optimizer.__exit__(None, None, None)
+
+        if os.environ.get("REUSE_DB") != "teardown":
+            # skip remaining teardown
```

Comment:
What if `REUSE_DB` isn't set?  It might be safer to always teardown unless `REUSE_DB in ("1", "reset", "optimize")`